File: src/main/java/com/DavidM1A2/AfraidOfTheDark/common/entities/spell/projectile/AnimationHandlerSpellProjectile.java
/*
 * Author: David Slovikosky
 * Mod: Afraid of the Dark
 * Ideas and Textures: Michael Albertson
 */
package com.DavidM1A2.AfraidOfTheDark.common.entities.spell.projectile;

import java.util.HashMap;

import com.DavidM1A2.AfraidOfTheDark.common.MCACommonLibrary.IMCAnimatedEntity;
import com.DavidM1A2.AfraidOfTheDark.common.MCACommonLibrary.animation.AnimationHandler;
import com.DavidM1A2.AfraidOfTheDark.common.MCACommonLibrary.animation.Channel;

public class AnimationHandlerSpellProjectile extends AnimationHandler
{
	/** Map with all the animations. */
	public static HashMap<String, Channel> animChannels = new HashMap<String, Channel>();

	static
	{
		animChannels.put("Idle", new ChannelSpellProjectileIdle("Idle", 100.0F, 60, Channel.LOOP));
	}

	public AnimationHandlerSpellProjectile(IMCAnimatedEntity entity)
	{
		super(entity);
	}

	@Override
	public void activateAnimation(String name, float startingFrame)
	{
		super.activateAnimation(animChannels, name, startingFrame);
	}

	@Override
	public void stopAnimation(String name)
	{
		super.stopAnimation(animChannels, name);
	}
}

File: src/main/java/com/davidm1a2/afraidofthedark/common/entity/spell/projectile/animation/AnimationHandlerSpellProjectile.kt
package com.davidm1a2.afraidofthedark.common.entity.spell.projectile.animation

import com.davidm1a2.afraidofthedark.common.entity.mcAnimatorLib.animation.AnimationHandler
import com.davidm1a2.afraidofthedark.common.entity.mcAnimatorLib.animation.ChannelMode

/**
 * Animation handler class for the spell projectile
 */
class AnimationHandlerSpellProjectile : AnimationHandler(
    setOf(
        ChannelSpellProjectileIdle("Idle", 100.0f, 60, ChannelMode.LOOP)
    )
)